services/QuillLMS/spec/controllers/teachers/units_controller_spec.rb

```diff
@@ -384,6 +386,23 @@
       } }
       expect(response.status).to eq(422)
     end
+
+    it 'passes only visible unit activities to the updater' do
+      classroom_data = [{ id: classroom.id.to_s, student_ids: [student.id.to_s] }]
+
+      expect(Units::Updater).to receive(:run).with(
+        unit.id.to_s,
+        [{ id: unit_activity.activity_id }],
+        classroom_data,
+        teacher.id
+      )
```

Comment:
I kind of want a test expectation that captures that we definitely don't want to get the `non_visible_unit_activity` passed in for updating, but I actually don't know of a clean way of doing that given the current code, so this really does seem like the best test in this spot.